boot-qemu.py

```diff
@@ -335,13 +397,17 @@ def parse_arguments():
                 f"EFI boot requested on unsupported architecture ('{args.architecture}'), ignoring...",
             )
 
+    if args.gdb:
+        runner.gdb = True
+        runner.gdb_bin = args.gdb_bin
+
     if args.no_kvm:
         runner.use_kvm = False
 
     if args.smp:
         runner.smp = args.smp
 
-    runner.interactive = args.interactive
+    runner.interactive = args.interactive or args.gdb
```

Comment:
Let's keep these separate for the runner. i.e. `gdb` should not imply `interactive`. Probably need to adjust the conditional that sets `'rdinit=/bin/sh'` above?

Reply:
Ack, it is not necessary to do this globally now that we have separate paths for running QEMU in the foreground versus background with GDB because we add `timeout` and `stdbuf` only on the foreground path:

09a59ca6be112027bfac117c325c35847eb6272d